mesonbuild/modules/gnome.py

```diff
@@ -553,15 +553,16 @@ def _scan_gir_targets(self, state, girtargets):
             else:
                 # Because of https://gitlab.gnome.org/GNOME/gobject-introspection/merge_requests/72
                 # we can't use the full path until this is merged.
+                libpath = os.path.join('/', girtarget.get_subdir(), girtarget.get_filename())
                 if isinstance(girtarget, build.SharedLibrary):
+                    # need to put our output directory first as we need to use the
+                    # generated libraries instead of any possibly installed system/prefix
+                    # ones.
+                    ret += ["-L@BUILD_ROOT@%s" % os.path.dirname(libpath)]
```

Comment:
Do not start `libpath` with `/`. It is very confusing because it looks like you are building an absolute path instead. Have it in the format string instead. Text formatting should also be done with format rather than `%` as we are trying to phase out the latter.

Thus this should be something like:

```meson
libpath = os.path.join(girtarget.get_subdir(), girtarget.get_filename())
# and
ret += ["-L@BUILD_ROOT@/%s".format(os.path.dirname(libpath))]
```

Reply:
Thanks, that is better.